include/shardwrapper.hpp

```diff
@@ -153,10 +154,17 @@ template <class T> struct ShardWrapper {
 
     // setParam
     if constexpr (has_setParam<T>::value) {
-      result->setParam = static_cast<SHSetParamProc>(
-          [](Shard *b, int i, const SHVar *v) { reinterpret_cast<ShardWrapper<T> *>(b)->shard.setParam(i, *v); });
+      result->setParam = static_cast<SHSetParamProc>([](Shard *b, int i, const SHVar *v) {
+        try {
+          reinterpret_cast<ShardWrapper<T> *>(b)->shard.setParam(i, *v);
+          return SHError{0};
```

Comment:
I feel like for clarity we should have a static `constexpr` like `SHError::Success` to replace all the instances of `SHError{0}`.

Reply:
True, for now it's fine but once we do the big refactor to remove exceptions might be better to have something more clear.

Reply:
Sorted